File: src/main/kotlin/com/natu/ftax/transaction/domain/TransactionType.java
package com.natu.ftax.transaction.domain;

import java.util.Locale;

enum TransactionType {
    TRANSFER, SWAP, NONE;

    static TransactionType forValue(String value) {
        return TransactionType.valueOf(value.toUpperCase(Locale.getDefault()));
    }
}

File: src/main/kotlin/com/natu/ftax/transaction/domain/TransactionType.kt
package com.natu.ftax.transaction.domain

import com.fasterxml.jackson.annotation.JsonCreator
import java.util.*

enum class TransactionType {
    TRANSFER, SWAP, NONE;

    companion object {
        @JsonCreator
        @JvmStatic
        fun forValue(value: String): TransactionType =
            TransactionType.valueOf(value.uppercase(Locale.getDefault()))
    }
}
